cmd/query/app/query_parser.go

```diff
@@ -76,15 +93,15 @@ type traceQueryParameters struct {
 //     key := strValue
 //     keyValue := strValue ':' strValue
 //     tags :== 'tags=' jsonMap
-func (p *queryParser) parse(r *http.Request) (*traceQueryParameters, error) {
+func (p *queryParser) parseTraceQueryParams(r *http.Request) (*traceQueryParameters, error) {
 	service := r.FormValue(serviceParam)
 	operation := r.FormValue(operationParam)
 
-	startTime, err := p.parseTime(startTimeParam, r)
+	startTime, err := p.parseTime(r, startTimeParam, time.Microsecond)
```

Comment:
the `search` endpoint defines times in microseconds since epoch (why?), but other endpoints such as `dependencies` and `metrics` use milliseconds since epoch, since those are the time units JS (i.e. Jaeger UI) use e.g. `Date.now()`, so we need to pass the units to the `parseTime` function to support both cases.

Reply:
We should add this to comments in the code. The search operates on span latencies, which are expressed as microseconds in the data model, so it makes sense to support high accuracy. The search UI itself does not insist on exact units because it supports string like 1ms. We had a debate over whether units should be handled by the UI instead of the backend service, but here we are, since Go makes parsing 1ms very easy.

The dependencies API does not operate on the latency space, instead its timestamps are just time range selections, and the typical backend granularity of those is on the order of 15min or more. So microseconds aren't really that useful in this domain, although I certainly would've preferred having a consistent time representation.

Metrics is a new domain, you need to decide which representation makes more sense. I think it's closer to dependencies, where you're asking about wall clock time, not the latency domain.

Reply:
> The search operates on span latencies, which are expressed as microseconds in the data model

Out of curiosity, why micros and not posix nanos which I think also fit into an int64? Were there some limitations in storage or clients to handle posix nanos?

> Metrics is a new domain, you need to decide which representation makes more sense. I think it's closer to dependencies, where you're asking about wall clock time, not the latency domain.

I think milliseconds are most appropriate (I've documented these reasons in comments within `queryParser`):
- The main client in mind is Jaeger UI. Being a react.js app, its built-in time precision is milliseconds so it simplifies usage for Jaeger UI.
- The min step size is `1ms` for the Prometheus-compliant storage backends I've tested.
- It also follows that data point timestamps are reported with a millisecond precision, at least from my tests so far.

Reply:
> Out of curiosity, why micros and not posix nanos which I think also fit into an int64? Were there some limitations in storage or clients to handle posix nanos?

legacy from zipkin origins, which still only supports micros, whereas our domain model uses Go's time.Time which has nanos. But our Thrift API uses micros.